pubmedcentral/download_and_convert_to_md.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import functools
+import multiprocessing as mp
+import os
+import tarfile
+
+from tqdm import tqdm
+
+cur_dir = os.getcwd()
+
+parser = argparse.ArgumentParser(description="Convert xml documents to markdown.")
+parser.add_argument("--filelist", help="The path to the filelist.txt file.")
+parser.add_argument(
+    "--output_dir", default="data/raw/", help="Where the markdown files go."
+)
+parser.add_argument("--total_docs", default=0, type=int, help="Total number of documents to convert, for debugging.")
+
+BASE_URL = "https://ftp.ncbi.nlm.nih.gov/pub/pmc/"
+
+
+def download(f_url: str, output_dir: str):
+    # download file from f_url to output_dir
+    try:
+        # -nc: no clobber, -q: quiet
+        os.system(f"wget -nc -q {f_url} -P {output_dir}")
+    except:
+        print(f"Error downloading {f_url}")
+        import traceback
+
+        traceback.print_exc()
+
+
+def extract_and_convert_tarball(t: str, output_dir: str):
+    if not os.path.exists(t):
+        return
+    try:
+        with tarfile.open(t) as tar:
+            files = tar.getnames()
+            nxml = [f for f in files if f.endswith(".nxml")]
+
+            # make sure there's only one nxml file
```

Comment:
How often is there more that one xml file? Does it mean their data is messed up or is it an edgecase we don't handle? If the latter, is it worth handling?

Reply:
Done